src/cargo/core/resolver/mod.rs

```diff
@@ -1091,12 +1088,9 @@ fn activate_deps_loop(
                             .filter_map(|(_, (deb, _, _))| {
                                 past_conflicting_activations.get(&deb).and_then(|past_bad| {
```

Comment:
This may actually be a great place to use `?`:

```rust
past_conflicting_activations.get(&deb)?
    .iter()
    .find(|conflicts| cx.is_conflicting(None, conflicts))
```

(also this may want to rename `deb` to `dep`

Reply:
So much better!